### PR TITLE
Leila / Ram metric 46

### LINKED ISSUE #46: Sprint 2: Add RAM metric collection

### PR DESCRIPTION
**Pull request by l3331l4**
RAM functions tested and working, added method calls to exporter and updated Prometheus config to read in metric data.

Unit tests written for each method.

Closes #46.

### CODE REVIEW

**Comment by max-romagnoli** [comment]
@ljdzed attention needed on this PR.
Need to request change to call method in exporter.py and metrics.py.
Let's try to close it soon, adding at least one or two tests and seeing if Prometheus reads the metrics exposed.

**Review by ljdzed — CHANGES_REQUESTED** [review]
@max-romagnoli Confirm with Max that prometheus.yml is merged correctly before approval. 

Otherwise good build and good tests added.

**Comment by max-romagnoli** [comment]
@l3331l4 
Everything looks good. Great job on the tests and connecting your metrics to Prometheus!
This finally closes #15.

I merged dev into this branch to solve any conflict. So I would pull these changes locally and test it yourself with the new docker compose setup (have a look at the `REAMDE_Docker`).

File: exporter/metrics.py [hunk 1]
```diff
@@ -2,4 +2,7 @@
 
 cpu_utilization_gauge = Gauge("cpu_utilization", "CPU utilization percentage")
 cpu_frequency_gauge = Gauge("cpu_frequency", "CPU frequency in MHz")
-cpu_temperature_gauge = Gauge("cpu_temperature", "CPU temperature in Celsius")
\ No newline at end of file
+cpu_temperature_gauge = Gauge("cpu_temperature", "CPU temperature in Celsius")
+
+ram_utilization_gauge = Gauge("ram_utilization", "RAM utilization percentage")
+ram_memory_gauge = Gauge("ram_memory", "Total RAM memory installed")
```
Comment (max-romagnoli) [inline exporter/metrics.py:8]: ## Requested Changes
* We need to make the Gauges descriptions more comprehensive. We should include some information on the unit of measure and prefix all of them with the name of our exporter (say `stagent_`). For example:
```suggestion
ram_memory_gauge = Gauge("stagent_ram_memory_bytes", "Total RAM memory installed in bytes")
```

**Review by max-romagnoli — CHANGES_REQUESTED** [review]

**Review by QimingNie — APPROVED** [review]

**Review by QimingNie — DISMISSED** [review]

**Comment by l3331l4** [comment]
Tested and working on new docker setup, improved gauge descriptions for RAM metrics.

**Review by max-romagnoli — APPROVED** [review]
# LGTM

**Pull request merged**